config/base/derive/src/lib.rs

```diff
@@ -500,3 +500,277 @@ fn impl_configurable(ast: &DeriveInput) -> TokenStream {
+            let meta = attr
+                .parse_meta()
+                .expect("derive macro must be in one of the meta forms");
+            match meta {
+                Meta::List(list) => {
+                    let items: Vec<syn::NestedMeta> = list
+                        .nested
+                        .into_iter()
+                        .filter(|nested| {
+                            if let NestedMeta::Meta(Meta::Path(path)) = nested {
+                                if path.is_ident("Default") {
+                                    return false;
+                                }
+                            }
+                            true
+                        })
+                        .collect();
+                    *attr = syn::parse_quote!(
+                        #[derive(#(#items),*)]
+                    )
+                }
+                Meta::Path(path) if path.is_ident("Default") => {
+                    *attr = syn::parse_quote!(
+                        #[derive()]
+                    )
+                }
+                _ => {}
+            }
+        }
+    }
+    remove_attr_struct(&mut ast, "view");
+    ast.ident = format_ident!("{}View", ast.ident);
+    ast
+}
+
+fn gen_impl_from(original: &ViewInput, view: &ViewInput) -> proc_macro2::TokenStream {
+    let ViewInput {
+        ident: original_ident,
+        ..
+    } = original;
+    let ViewInput {
+        generics,
+        ident: view_ident,
+        fields,
+        ..
+    } = view;
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+    let field_idents = extract_field_idents(fields);
+
+    quote! {
+        impl #impl_generics From<#original_ident> for #view_ident #ty_generics #where_clause {
+            fn from(config: #original_ident) -> Self {
+                let #original_ident {
+                    #(
+                        #field_idents,
+                    )*
+                    ..
+                } =  config;
+                Self {
+                    #(
+                        #field_idents: From::<_>::from(#field_idents),
+                    )*
+                }
+            }
+        }
+    }
+}
+
+fn gen_impl_default(original: &ViewInput, view: &ViewInput) -> proc_macro2::TokenStream {
+    let ViewInput {
+        ident: original_ident,
+        ..
+    } = original;
+    let ViewInput {
+        generics,
+        ident: view_ident,
+        fields,
+        ..
+    } = view;
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+    let field_idents = extract_field_idents(fields);
+
+    quote! {
+        impl #impl_generics Default for #view_ident #ty_generics #where_clause {
+            fn default() -> Self {
+                let #original_ident {
+                    #(
+                        #field_idents,
+                    )*
+                    ..
+                } =  #original_ident::default();
```

Comment:
```suggestion
                } =  <#original_ident as Default>::default();
```

This may be important if someone will write `default()` method for his structure not as `Default` impl

Reply:
Thanks to your comment i rethink `Default` implementation, it could be as simple as:
```rust
fn default() -> Self {
    Self::from(<#original_ident as Default>::default())
}
```